test/jdk/java/nio/channels/AsyncCloseAndInterrupt.java

```diff
@@ -469,8 +469,9 @@ private void caught(Channel ch, IOException x) {
                     SocketChannel sc = (SocketChannel)ch;
                     if (!sc.socket().isOutputShutdown())
                         throw new RuntimeException("Output not shutdown");
-                } else if ((test == TEST_INTR) && (op == TRANSFER_FROM)) {
-                    // Let this case pass -- CBIE applies to other channel
+                } else if ((test == TEST_INTR || test == TEST_PREINTR)
+                        && (op == TRANSFER_FROM)) {
+                    // Let these cases pass -- CBIE applies to other channel
```

Comment:
Would it be possible to explain why this didn't happen before? The file channel position is 0 and I think transferFrom is called to transfer up to 1MB from a pipe source. The method is called with the interrupt status set so attempting from the pipe source will throw ClosedByInterruptException. I'm wondering what this didn't happen before your change.

Reply:
transferFrom() simply returned and so never got to reading from the source.